plan/relations.go

```diff
@@ -607,14 +607,14 @@ func (j *JoinRel) RecordType() types.StructType {
 	switch j.joinType {
```

Comment:
Are you going to need JoinTypeRightSingle  and JoinTypeRightAnti for all of the options to be handled without a default clause?

Reply:
This PR I intended to keep for proto change so I planned to handle them in a separate PR. I updated PR to add missing ones to indicate unsupported. Kindly check